### ISSUE TITLE
make font size persisting to settings optional

### ISSUE THREAD

**Issue by aexvir** [zed-industries/zed#23505]
### Summary 

Persisting font size changes to settings.json in https://github.com/zed-industries/zed/pull/23265 removes the ability to have a custom default font size / regresses the behavior of "Reset Zoom" (cmd-0).

### Description

* most of my font size changes are temporary and done on the spot to fulfill a purpose
  * either because I'm screensharing and I need to zoom in
  * or because I'm laying in my chair more than usual so I need to see stuff bigger for a while
* I have my ideal font size that I always can rely on and reset to via cmd+0
  * now cmd+0 sets the size to null, which means some default I don't have control over
* I have my settings file tracked in git
  * I don't want my (mostly temporary) adjustments on font size to end up triggering changes in my dotfiles

I'd like if there were some way to support the old behaviour.

### Zed Version and System Specs

Zed: v0.171.0 (Zed Preview)
OS: macOS 15.2.0
Memory: 18 GiB
Architecture: aarch64

**Comment by notpeter** [comment]
@osiewicz

**Comment by notpeter** [comment]
The fact that cmd-0 no longer allows one to have "base" font size which differs from the Zed default buffer_font_size of 15 (unless you never zoom) feels like a regression.

**Comment by didier** [comment]
Second this. I really liked the font size feeling "temporary" and being able to reset it.

**Comment by paveldroo** [comment]
Totally agree, when you are on a call and just temporary zooming in for explaining some parts of code, you should be able to return to your standard font size to continue coding after a call.
This is default behavior of terminals + nvim as a part of terminal for sure. And default routine on calls zoom-in/back to normal. A little unexpected that cmd+0 now does nothing useful, just setting "null" for the font size.

**Comment by injust** [comment]
I'm used to using VS Code and this change caught me by surprise.

Was there any precedent for this? I'm curious because I can't recall using an editor that behaved like this.

**Comment by Dot32Dev** [comment]
> I'm used to VS Code and this change caught me by surprise. I'm curious if there was any precedent for this, because I can't recall using an editor that behaved this way.
> 
> I don't see the point of making this configurable, either. If someone could find it in the settings, wouldn't they rather modify `buffer_font_size` instead?

Sublime Text saves your zoom to your config like this, but I agree that it makes more sense to just change your preferred default size in settings. Sublime also lags when zooming due to having to update the file before the changes appeared. I liked being able to temporarily zoom with the keyboard in Zed, as it was very useful when explaining code or concepts to other people.

**Comment by injust** [comment]
This wasn't completely fixed, see https://github.com/zed-industries/zed/issues/34479.
